url3010dbde70_nbartlett_Sequence_Memoizer_tgz-pJ8-edu_columbia_stat_wood_pub_sequencememoizer_IntSequenceMemoizerJ8/src/edu/columbia/stat/wood/pub/sequencememoizer/IntSequenceMemoizer.java

```diff
@@ -0,0 +1,846 @@
+
+        isn.sample();
+        isn.fillRestaurant(r);
+    }
+
+    private double sampleDiscounts(double stdProposal) {
+        double logLik, pLogLik, currentValue, proposal;
+        boolean accept;
+
+        logLik = score();
+
+        for (int dIndex = 0; dIndex < discounts.length(); dIndex++) {
+            currentValue = discounts.get(dIndex);
+            proposal = currentValue + stdProposal * RNG.nextGaussian();
+
+            if (proposal > 0.0 && proposal < 1.0) {
+                discounts.set(dIndex, proposal);
+                pLogLik = score();
+
+                accept = RNG.nextDouble() < Math.exp(pLogLik - logLik);
+                if (accept) {
+                    logLik = pLogLik;
+                } else {
+                    discounts.set(dIndex, currentValue);
+                }
+            }
+        }
+
+        currentValue = discounts.getdInfinity();
+        proposal = currentValue + stdProposal * RNG.nextGaussian();
+
+        if (proposal > 0.0 && proposal < 1.0) {
+            discounts.setDInfinity(proposal);
+            pLogLik = score();
+
+            accept = RNG.nextDouble() < Math.exp(pLogLik - logLik);
+            if (accept) {
+                logLik = pLogLik;
+            } else {
+                discounts.setDInfinity(currentValue);
+            }
+        }
+
+        return logLik;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public double score() {
+        double logLik;
+        int tti;
+
+        logLik = 0.0;
+        tti = 1;
+        for (int i = 0; i < ecr.types.length; i++) {
+            logLik += ecr.customersAndTables[tti] * Math.log(baseDistribution.probability(ecr.types[i]));
+            tti += 2;
+        }
+
+        return logLik += score(ecr, 0);
+    }
+
+    private double score(IntRestaurant r, int restaurantDepth) {
+        double logLik, discount;
+        LogGeneralizedSterlingNumbers lgsn;
+        int tci, tti;
+
+        logLik = 0.0;
+        if (!r.isEmpty()) {
+            for (Object child : r.values()) {
+                logLik += score((IntRestaurant) child, restaurantDepth + ((IntRestaurant) child).edgeLength);
+            }
+        }
+
+        discount = discounts.get(restaurantDepth - r.edgeLength, restaurantDepth);
+        lgsn = new LogGeneralizedSterlingNumbers(discount);
+
+        logLik += LogBracketFunction.logBracketFunction(discount, r.tables - 1, discount);
+        logLik -= LogBracketFunction.logBracketFunction(1, r.customers - 1, 1.0);
+
+        tci = 0;
+        tti = 1;
+        for (int i = 0; i < r.types.length; i++) {
+            logLik += lgsn.get(r.customersAndTables[tci], r.customersAndTables[tti]);
+            tci += 2;
+            tti += 2;
+        }
+
+        return logLik;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public IntSequenceMemoizerParameters getParameters() {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    private IntRestaurant getWithInsertion() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 48 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.